src/rules/no-descending-specificity/index.js

```diff
@@ -51,18 +51,23 @@ export default function (actual) {
         comparisonContext.set(lastNonPseudoSelectorNode, [entry])
         return
       }
-
+      // Return early if selector contains a matches
+      if (selector.indexOf(":matches(") !== -1) { return }
       const priorComparableSelectors = comparisonContext.get(lastNonPseudoSelectorNode)
 
       priorComparableSelectors.forEach(priorEntry => {
-        if (compare(selectorSpecificity, priorEntry.specificity) === -1) {
-          report({
-            ruleName,
-            result,
-            node: rule,
-            message: messages.rejected(selector, priorEntry.selector),
-            index: sourceIndex,
-          })
+        try {
+          if (compare(selectorSpecificity, priorEntry.specificity) === -1) {
+            report({
+              ruleName,
+              result,
+              node: rule,
+              message: messages.rejected(selector, priorEntry.selector),
+              index: sourceIndex,
+            })
+          }
+        } catch (e) {
+          result.warn("Cannot parse selector", { e })
```

Comment:
Checked https://github.com/stylelint/stylelint/blob/0f0056003738d11a100d7b02da45f2cdec99b5f6/src/utils/parseSelector.js#L7 --- that object should include the node.